### PR TITLE
Add Joystick

### LINKED ISSUE #52: Add Joystick
As a user I want to control the car using a joystick instead of buttons

**Acceptance criteria:**
- [x] Have a visual representation of a joystick, in the shape of a circle 
- [x] Interacctive joystick where the user can move an inner circle to represent where the joystick has been moved to
- [x] When using the joystick it should be recognized by the MQTT broker, checking by sending a message to the serial command 
- [x] When interacting with left- and right side of the joystick (perpendicular to the y:axis) it should move the car to the left and right
- [x] When interacting with top- and bottom side of the joystick (perpendicular to the x:axis) it should move the car forwards and backwards

### PR DESCRIPTION
**Pull request by AlbinKarlsson**
## Created a joystick, a new steering mechanism for the app.  

Provides a smooth experience for the user driving the car. 
Close #52 

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change

### CODE REVIEW

File: android/app/src/main/java/platis/solutions/smartcarmqttcontroller/MainActivity.java [hunk 1]
```diff
@@ -363,8 +352,14 @@ public void mqttConnectionStatus(boolean isConnected){
 
     }
 
-}
-
 
+    @Override
```
Comment (Lindet94) [inline android/app/src/main/java/platis/solutions/smartcarmqttcontroller/MainActivity.java:357]: Can you add an explanation of this method?

File: arduino/smartcar/smartcar.ino [hunk 2]
```diff
@@ -96,8 +96,10 @@ void setup()
       {
     if (topic == "/smartcar/control/throttle") {
         car.setSpeed(message.toInt());
+        Serial.println(message + "Felix " );
```
Comment (Lindet94) [inline arduino/smartcar/smartcar.ino]: Why print Felix here?

File: arduino/smartcar/smartcar.ino [hunk 3]
```diff
@@ -96,8 +96,10 @@ void setup()
...
     } else if (topic == "/smartcar/control/steering") {
         car.setAngle(message.toInt());
+        Serial.println(message + "Albin " );
```
Comment (Lindet94) [inline arduino/smartcar/smartcar.ino]: Why print Albin here?

**Review by Lindet94 — CHANGES_REQUESTED** [review]
Checked code and it looks good. Just a couple if prints that I'm not sure about why they are included.

**Review by AlbinKarlsson — COMMENTED** [review]
Fixed all of Eriks requests

**Review by Lindet94 — APPROVED** [review]

**Comment by Lindet94** [comment]
Good changes.

**Pull request merged**